src/javascript/ContentEditor.redux.jsx

```diff
@@ -17,18 +20,42 @@ const mapStateToProps = state => {
 
 const ContentEditorReduxCmp = ({client, mode, uuid, lang, uilang, site, contentType}) => {
     const {redirect, hasHistory, exit, registerBlockListener, unRegisterBlockListener} = useContentEditorHistory();
+    const {storedLocation, setStoredLocation} = useContentEditorHistoryContext();
+    const {openPaths} = useSelector(state => ({openPaths: state.jcontent.openPaths}));
+    const dispatch = useDispatch();
     const {t} = useTranslation();
     // Sync GWT language
     if (window.top.authoringApi.switchLanguage) {
         window.top.authoringApi.switchLanguage(lang);
     }
 
+    const rename = (node, mutateNode) => {
+        if (mutateNode && mutateNode.rename) {
+            if (storedLocation && storedLocation.location) {
+                if (openPaths) {
+                    dispatch(replaceOpenedPath(openPaths.map(openPath => openPath.replace(node.path, mutateNode.rename))));
+                }
+
+                const {site: currentSite, language, mode: currentMode, path} = registry.get('jcontent', 'utils').extractParamsFromUrl(storedLocation.location.pathname, storedLocation.location.search);
```

Comment:
### ISSUE 1:
you pass the storedLocation.location.search but you never get the associated params, so the rebuilt url will not have the original search querystring if there is one.
Scenario:
- open Jcontent
- search for a content
- edit the current folder and rename
- do back
- you should get back to the renamed folder with the previous search (currently the search is lost)